exporter/collector/metrics.go

```diff
@@ -68,12 +68,21 @@ type MetricsExporter struct {
 	shutdownC chan struct{}
 	// mdCache tracks the metric descriptors that have already been sent to GCM
 	mdCache map[string]*monitoringpb.CreateMetricDescriptorRequest
-	cfg     Config
+	// requestOpts applies options to the context for requests, such as additional headers.
+	requestOpts []func(*context.Context, requestInfo)
+	mapper      metricMapper
+	cfg         Config
 	// goroutines tracks the currently running child tasks
 	goroutines sync.WaitGroup
 	timeout    time.Duration
 }
 
+// requestInfo is meant to abstract info from CreateMetricsDescriptorRequests and
+// CreateTimeSeriesRequests that is shared by requestOpts functions.
+type requestInfo struct {
+	name string
```

Comment:
maybe call this projectName or something more descriptive?

Reply:
Sure. Tbh I'm not super happy with this approach in the first place, but it was the best way I could think of to re-use the functionality across the different metric export functions. I'm open to other architectural suggestions for this, but otherwise I did the rename. This could be useful in the future if we want to do other modifications before sending the request

Reply:
maybe `type requestProject string`?  I'm fine with whatever you think is best.